back/app/models/permission.rb

```diff
@@ -59,6 +60,10 @@ def set_permitted_by_and_global_custom_fields
     self.permitted_by ||= 'users'
     self.global_custom_fields ||= (permitted_by == 'users')
   end
+
+  def update_global_custom_fields
+    self.global_custom_fields = false if permitted_by == 'everyone_confirmed_email'
```

Comment:
@jamesspeake `global_custom_fields` are still being set twice. On line 61 and on line 65.